compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleDependency.java

```diff
@@ -27,10 +27,13 @@
 public class ModuleDependency {
     private final PackageDependency packageDependency;
     private final ModuleId moduleId;
```

Comment:
Let's remove the moduleId if it is not needed now

Reply:
There is one usage in the `BalaWriter` class that does not use the dependency graph. I can change it to use `descriptor().name()` as well. https://github.com/ballerina-platform/ballerina-lang/blob/e6fbef887ce946b8e1bf50bc128d97f17e5416ba/compiler/ballerina-lang/src/main/java/io/ballerina/projects/BalaWriter.java#L330